R/utils.R

```
@@ -257,3 +257,243 @@ set_data_labels <-
+                                        part)
+        break
+      } else if (!grepl("::", part)) {
+        # This is the default variable (no database prefix)
+        default_var <- part
+      }
+    }
+    
+    if (!is.null(database_specific_match)) {
+      # Use the database-specific match - this is a direct database variable
+      var_name <- gsub("\\[|\\]", "", database_specific_match)
+      result_list <- append(result_list, list(list(name = var_name, 
+                                                    type = "database")))
+    } else if (!is.null(default_var)) {
+      # Use the default variable - this is also a database variable 
+      # (just the default for this database)
+      var_name <- gsub("\\[|\\]", "", default_var)
+      result_list <- append(result_list, list(list(name = var_name, 
+                                                    type = "database")))
+    } else {
+      stop("No matching variable found for database `", database, 
+           "` in variableStart `", variable_start, "`.", call. = FALSE)
+    }
+    
+  } else {
+    # Unrecognized format
+    stop("Unrecognized variableStart format: `", variable_start, "`.", 
+         call. = FALSE)
+  }
+  
+  return(result_list)
+}
+
+#' Helper function to determine variable type by looking it up in variables 
+#' sheet
+#' 
+#' @param var_name Character. Name of the variable to look up
+#' @param database Character. Database name for context
+#' @param variables_sheet Data frame. The variables sheet
+#' 
+#' @return Character. The type of the variable: "derived" or "non-derived"
+#' @keywords internal
+get_variable_type <- function(var_name, database, variables_sheet) {
+  # Find the variable in the sheet
+  var_rows <- variables_sheet[
+    variables_sheet[[pkg.env$columns.variable]] == var_name, ]
+  
+  if (nrow(var_rows) == 0) {
+    # If variable not found in sheet, assume it's non-derived 
+    # (could be a raw database variable)
+    return("non-derived")
+  }
```

Should throw an error instead since database variables are not allowed as starts for a derived variable